src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -2475,13 +2507,14 @@ void ColorControlServer::levelControlColorTempChangeCommand(EndpointId endpoint)
             currentLevel.SetNonNull((uint8_t) 0x7F);
         }
 
-        uint16_t tempPhysMax = MAX_TEMPERATURE_VALUE;
-        Attributes::ColorTempPhysicalMaxMireds::Get(endpoint, &tempPhysMax);
+        uint32_t tempCoupleMin = getTemperatureCoupleToLevelMin(endpoint);
+        uint32_t tempPhysMax   = MAX_TEMPERATURE_VALUE;
+        Attributes::ColorTempPhysicalMaxMireds::Get(endpoint, reinterpret_cast<uint16_t *>(&tempPhysMax));
 
         // Scale color temp setting between the coupling min and the physical max.
         // Note that mireds varies inversely with level: low level -> high mireds.
         // Peg min/MAX level to MAX/min mireds, otherwise interpolate.
-        uint16_t newColorTemp;
+        uint32_t newColorTemp;
```

Comment:
It seems like it would be clearer to get things into the narrower type here as soon as we can, so we're clear on which casts are actually happening and which are OK (e.g. we 're very sure that `getTemperatureCoupleToLevelMin` returns `uint16_t`, if we have the compiler enforce that....)

Reply:
Ok, I'll rework this. I am trying to limit the number of casts while still using a larger type for the calculation steps.